map.go

```diff
@@ -235,7 +235,29 @@ func (m *Map) Lookup(key, valueOut interface{}) error {
 	if valueBytes == nil {
 		return nil
 	}
+	return m.valueProcessing(valueOut, valueBytes)
+}
+
+// LookupAndDelete retrieves and deletes a value from a Map.
+//
+// Calls Close() on valueOut if it is of type **Map or **Program,
+// and *valueOut is not nil.
+//
+// Returns an error if the key doesn't exist, see IsNotExist.
+func (m *Map) LookupAndDelete(key, valueOut interface{}) error {
+	valuePtr, valueBytes := makeBuffer(valueOut, m.fullValueSize)
+
+	if err := m.lookup(key, valuePtr); err != nil {
+		return err
+	}
 
+	if valueBytes == nil {
+		return nil
+	}
+	return m.valueProcessing(valueOut, valueBytes)
+}
+
+func (m *Map) valueProcessing(valueOut interface{}, valueBytes []byte) error {
```

Comment:
I'm not sure about naming conventions here. As there needs to be the same post processing for values returned from the map for `Lookup()` and `LookupAndDelete()` and I wanted to avoid duplicating code.

Reply:
How about `unmarshalValue`?

Reply:
i've renamed it to `unmarshalValue`.